multicluster/apis/multicluster/v1alpha1/resourceexport_types.go

```diff
@@ -0,0 +1,112 @@
+/*
+Copyright 2021 Antrea Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"antrea.io/antrea/pkg/apis/crd/v1alpha2"
+)
+
+// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+// ServiceExport exports Service.
+type ServiceExport struct {
+	ServiceSpec v1.ServiceSpec `json:"serviceSpec,omitempty"`
+}
+
+// EndpointsExport exports Endpoints.
+type EndpointsExport struct {
+	Subsets []v1.EndpointSubset `json:"subsets,omitempty"`
+}
+
+// ExternalEntityExport exports ExternalEntity.
+type ExternalEntityExport struct {
+	ExternalEntitySpec v1alpha2.ExternalEntitySpec `json:"externalentityspec,omitempty"`
+}
+
+// NodeExport exports Node.
+type NodeExport struct {
```

Comment:
Question - in which case we need to export Node? Is it for NodePort Service?
Node is like cluster infra, so I am not very certain if we should export Node Status directly.

Reply:
if a user decides to add ServiceExport that refers to a local local Service of type NodePort, and Exporter will export some available Nodes IP (in NodeStatus)  externally so that the ServiceExport may be reached from outside of cluster.

Reply:
In that case should we export a separate struct similar to Endpoints or EndpointSlice (or even reuse them), rather than exporting each Node, esp. there are many fields in NodeStatus which are not needed at all?

And how you are going to handle ExternalPolicy = Local in this way? If we are not clear, I suggest to remove NodeExport from this PR first.

Reply:
@jianjuns I thought I am clear on what we need to do when ServiceExport uses NodePort, but I am not clear on ur questions about handing "ExternalPolicy = local", can u elaborate?